src/core/search.rs

```diff
@@ -238,14 +239,24 @@ pub trait FindexSearch<
         for (kwi, chain_table_uids) in kwi_chain_table_uids.iter() {
             let kwi_value: DemScheme::Key = kwi.derive_dem_key(CHAIN_TABLE_KEY_DERIVATION_INFO);
             let mut chain = Vec::with_capacity(chain_table_uids.len());
-            for uid in chain_table_uids {
-                // Fetch all chain table values one by one to increase noise.
-                let encrypted_item = self
-                    .fetch_chain_table(&HashSet::from([uid.clone()]))
-                    .await?;
+
+            // Fetch all chain table values one by one to increase noise.
+            let chain_table_uids_hashset: Vec<_> = chain_table_uids
+                .iter()
+                .map(|uid| HashSet::from([uid.clone()]))
+                .collect();
+
+            let mut futures = vec![];
```

Comment:
Can you initialize this with a capacity ? If there is not too many misses, the actual length should be close to `chain_table_uids_hashset.len()`.

Reply:
Done